cve_bin_tool/output_engine.py

```diff
@@ -0,0 +1,39 @@
+import csv
+
+from datetime import datetime
+from builtins import hash
+
+
+class OutputEngine:
+    def __init__(self, filename=None, modules=None):
+        self.filename = filename
+        self.modules = modules
+
+        if self.filename == None:
+            now = datetime.now()
+            self.filename = str(hash(now)).strip("-")
```

Comment:
I'm guessing the point of this to generate a somewhat-unique filename to avoid likely file collisions, but if we never tell the user what output we just wrote, I'm not sure if a `{lengthy numeric string}.csv` is going to be helpful, especially since once you run this more than once you're going to have a series of them.  

Some possible options:
1. Force the user to provide a filename and error if one is not provided (make it the user's problem)
2. Write an info level message so the user at least knows what filename was written
3. Add a prefix to the name so at least it's `output.cve-bin-tool.{numbers}.csv`
4. Use datetime to make a filename that gives you the date/time so at least the latest version is pretty obvious.

I'm thinking maybe 2, 3 *and* 4 would be a good combo here?
```python
self.filename = f"output.cve-bin-tool.{datetime.now().strftime('%Y-%m-%d.%H:%m:%S:%f')}.csv"
```
I can't decide if it's worth keeping the miliseconds (%f) in there for slightly reduced collision chance at expense of even longer filename.  We could add a short hash instead, but maybe that's over-thinking it.  I'm guessing if you have filename collisions at this level you should probably be generating your own filenames, so seconds is plenty for a default.